src/common/user-store/user-store.ts

```diff
@@ -109,7 +116,41 @@ export class UserStore extends BaseStore<UserStoreModel> /* implements UserStore
       });
     }, {
       fireImmediately: true,
-    });
+    }); 
+  }
+
+  // Returns monaco editor options for selected editor type (the place, where a particular instance of the editor is mounted)
+  getEditorOptions(type: EditorType): monaco.editor.IStandaloneEditorConstructionOptions{
+    return {
+      automaticLayout: true,
+      tabSize: UserStore.getInstance().editorConfiguration.get(type).tabSize,
+      minimap: UserStore.getInstance().editorConfiguration.get(type).miniMap,
+      lineNumbers: UserStore.getInstance().editorConfiguration.get(type).lineNumbers
+    };
+  }
+
+  getLineNumbers(type: EditorType): monaco.editor.LineNumbersType {
+    return UserStore.getInstance().editorConfiguration.get(type).lineNumbers;
+  }
+
+  setLineNumbers(type: EditorType, lineNumbers: monaco.editor.LineNumbersType) {
+    UserStore.getInstance().editorConfiguration.get(type).lineNumbers = lineNumbers;
+  }
+
+  getTabSize(type: EditorType): number {
+    return UserStore.getInstance().editorConfiguration.get(type).tabSize;
+  }
+
+  setTabSize(type: EditorType, tabSize: number) {
+    UserStore.getInstance().editorConfiguration.get(type).tabSize = tabSize;
+  }
+
+  isMinimapEnabled(type: EditorType): boolean {
+    return UserStore.getInstance().editorConfiguration.get(type).miniMap.enabled;
+  }
+
+  enableMinimap(type: EditorType, miniMap: boolean) {
+    UserStore.getInstance().editorConfiguration.get(type).miniMap.enabled = miniMap;
   }
```

Comment:
I think these can probably be reduced to a `getEditorConfiguration` function.

Furthermore, there is no need to call `UserStore.getInstance()` within `UserStore` methods. Just use `this`.